index.ts

```diff
@@ -148,110 +151,118 @@ async function init() {
           validate: (dir) => isValidPackageName(dir) || 'Invalid package.json name'
         },
         {
           name: 'needsTypeScript',
           type: () => (isFeatureFlagsUsed ? null : 'toggle'),
-          message: 'Add TypeScript?',
+          message: language.needsTypeScript.message,
           initial: false,
-          active: 'Yes',
-          inactive: 'No'
+          active: language.needsTypeScript.toggleOptions[0],
+          inactive: language.needsTypeScript.toggleOptions[1]
         },
         {
           name: 'needsJsx',
           type: () => (isFeatureFlagsUsed ? null : 'toggle'),
-          message: 'Add JSX Support?',
+          message: language.needsJsx.message,
           initial: false,
-          active: 'Yes',
-          inactive: 'No'
+          active: language.needsJsx.toggleOptions[0],
+          inactive: language.needsJsx.toggleOptions[1]
         },
         {
           name: 'needsRouter',
           type: () => (isFeatureFlagsUsed ? null : 'toggle'),
-          message: 'Add Vue Router for Single Page Application development?',
+          message: language.needsRouter.message,
           initial: false,
-          active: 'Yes',
-          inactive: 'No'
+          active: language.needsRouter.toggleOptions[0],
+          inactive: language.needsRouter.toggleOptions[1]
         },
         {
           name: 'needsPinia',
           type: () => (isFeatureFlagsUsed ? null : 'toggle'),
-          message: 'Add Pinia for state management?',
+          message: language.needsPinia.message,
           initial: false,
-          active: 'Yes',
-          inactive: 'No'
+          active: language.needsPinia.toggleOptions[0],
+          inactive: language.needsPinia.toggleOptions[1]
         },
         {
           name: 'needsVitest',
           type: () => (isFeatureFlagsUsed ? null : 'toggle'),
-          message: 'Add Vitest for Unit Testing?',
+          message: language.needsVitest.message,
           initial: false,
-          active: 'Yes',
-          inactive: 'No'
+          active: language.needsVitest.toggleOptions[0],
+          inactive: language.needsVitest.toggleOptions[1]
         },
         {
           name: 'needsE2eTesting',
           type: () => (isFeatureFlagsUsed ? null : 'select'),
-          message: 'Add an End-to-End Testing Solution?',
+          message: language.needsE2eTesting.message,
           initial: 0,
           choices: (prev, answers) => [
-            { title: 'No', value: false },
             {
-              title: 'Cypress',
+              title: language.needsE2eTesting.selectOptions[0].title,
+              value: false
+            },
+            {
+              title: language.needsE2eTesting.selectOptions[1].title,
               description: answers.needsVitest
                 ? undefined
-                : 'also supports unit testing with Cypress Component Testing',
+                : language.needsE2eTesting.selectOptions[1].desc,
```

Comment:
Again, array indexes are hard to understand and maintain (when we need to sync modifications among different languages). Please use more semantic keys.